app/page.tsx

```diff
@@ -11,21 +12,40 @@ import Link from "next/link";
 
 import { MenuButton } from "@/components/ui/menubutton.tsx";
 
-function getASRouterLocalColumnFromJSON(messageDef: any) : FxMSMessageInfo {
+async function getASRouterLocalColumnFromJSON(messageDef: any) : Promise<FxMSMessageInfo> {
   let fxmsMsgInfo : FxMSMessageInfo = {
     product: 'Desktop',
     id: messageDef.id,
     template: messageDef.template,
     surface: getDisplayNameForTemplate(getTemplateFromMessage(messageDef)),
     segment: 'some segment',
     metrics: 'some metrics',
-    ctrPercent: .5, // getMeFromLooker
+    ctrPercent: 0, // getMeFromLooker
     ctrPercentChange: 2, // getMeFromLooker
     previewLink: getPreviewLink(messageDef),
   };
 
-  fxmsMsgInfo.ctrDashboardLink = getDashboard(messageDef.template, messageDef.id,
-    "release")
+  let dbElement = await getAWDashboardElement0();
+
+  // console.log("dbElement.query: ", dbElement.query)
+  // console.log("dbElement.query.filters: ", dbElement.query.filters)
+  // console.log("dbElement.query.model: ", dbElement.query.model)
+  // console.log("dbElement.query.filter_expression: ",
+  //   dbElement.query.filter_expression)
+
+  const queryResult = await runEventCountQuery(
+    { 'event_counts.message_id':  '%' + messageDef.id + '%' }
+  )
+
+  // console.log("queryResult: ", queryResult)
+  if (queryResult.length > 0 && fxmsMsgInfo.template !== 'infobar') {
+    fxmsMsgInfo.ctrPercent = Number(Number(queryResult[0].primary_rate * 100).toFixed(1))
+  }
+  fxmsMsgInfo.ctrDashboardLink = getDashboard(messageDef.template, messageDef.id, "release")
+
+  // dashboard link -> dashboard id -> query id -> query -> ctr_percent_from_lastish_day
+
+  // console.log("fxmsMsgInfo: ", fxmsMsgInfo)
```

Comment:
I think all the changed lines above should be hoisted into their own function, since we're effective copy-pasting this code three times inside this file.  Whether or not the function wants to be massaged so that it mostly lives in looker.ts, I'm not sure.  My guess is that there's not a lot of win to doing that right this moment, but maybe that depends on testability.

Reply:
For now I have the logic inside its own function and moved to `looker.ts` so that I can try testing it along with the other looker functions